// @ainosoft/appops-core-components/components/enterprise-meta-config/dist/enterprise-meta-config/esm5/lib/enterprise-meta-config.module.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/enterprise-meta-config.module.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { NgModule, CUSTOM_ELEMENTS_SCHEMA, NO_ERRORS_SCHEMA } from '@angular/core';
import { EnterpriseMetaConfigComponent } from './enterprise-meta-config.component';
import { EnterpriseMetaConfigService } from './enterprise-meta-config.service';
import { MaterialModuleSet } from "../common/material-module";
import { CommonModule } from '@angular/common';
var EnterpriseMetaConfigModule = /** @class */ (function () {
    function EnterpriseMetaConfigModule() {
    }
    EnterpriseMetaConfigModule.decorators = [
        { type: NgModule, args: [{
                    declarations: [EnterpriseMetaConfigComponent],
                    imports: [
                        MaterialModuleSet,
                        CommonModule,
                    ],
                    exports: [EnterpriseMetaConfigComponent],
                    providers: [EnterpriseMetaConfigService],
                    schemas: [
                        CUSTOM_ELEMENTS_SCHEMA, NO_ERRORS_SCHEMA
                    ],
                },] }
    ];
    return EnterpriseMetaConfigModule;
}());
export { EnterpriseMetaConfigModule };
