// @ainosoft/appops-core-components/components/enterprise-meta-config/dist/enterprise-meta-config/esm5/lib/enterprise-meta-config.service.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/enterprise-meta-config.service.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Injectable } from '@angular/core';
import { MetaConfig } from '../config/MetaConfig';
import { BehaviorSubject } from 'rxjs';
import * as i0 from "@angular/core";
import * as i1 from "../config/MetaConfig";
var EnterpriseMetaConfigService = /** @class */ (function () {
    function EnterpriseMetaConfigService(metaConfigComp) {
        this.metaConfigComp = metaConfigComp;
        this.attrGroupListSource = new BehaviorSubject(this.attributeGroupList);
        this.currentAttrGroupList = this.attrGroupListSource.asObservable();
        this.attrOptionListSource = new BehaviorSubject(this.attributeOptionList);
        this.currentAttrOptionList = this.attrOptionListSource.asObservable();
        this.enableFormSource = new BehaviorSubject(this.enableForm);
        this.currentEnableFormValue = this.enableFormSource.asObservable();
    }
    /**
     * @param {?} value
     * @return {?}
     */
    EnterpriseMetaConfigService.prototype.enableOpenFormAction = /**
     * @param {?} value
     * @return {?}
     */
    function (value) {
        this.enableForm = value;
        this.enableFormSource.next(this.enableForm);
    };
    /**
     * @param {?} metaConfigHeader
     * @return {?}
     */
    EnterpriseMetaConfigService.prototype.setMetaConfigHeader = /**
     * @param {?} metaConfigHeader
     * @return {?}
     */
    function (metaConfigHeader) {
        this.metaConfigComp.setMetaConfigHeader = metaConfigHeader;
    };
    /**
     * @param {?} data
     * @return {?}
     */
    EnterpriseMetaConfigService.prototype.setAttrGroupList = /**
     * @param {?} data
     * @return {?}
     */
    function (data) {
        this.attributeGroupList = new Array();
        for (var i = 0; i < data.length; i++) {
            this.attributeGroupList.push(data[i]);
        }
        this.attrGroupListSource.next(this.attributeGroupList);
    };
    /**
     * @param {?} data
     * @return {?}
     */
    EnterpriseMetaConfigService.prototype.setAttrOptionList = /**
     * @param {?} data
     * @return {?}
     */
    function (data) {
        this.attributeOptionList = new Array();
        for (var i = 0; i < data.length; i++) {
            this.attributeOptionList.push(data[i]);
        }
        this.attrOptionListSource.next(this.attributeOptionList);
    };
    EnterpriseMetaConfigService.decorators = [
        { type: Injectable, args: [{
                    providedIn: 'root'
                },] }
    ];
    /** @nocollapse */
    EnterpriseMetaConfigService.ctorParameters = function () { return [
        { type: MetaConfig }
    ]; };
    /** @nocollapse */ EnterpriseMetaConfigService.ngInjectableDef = i0.ɵɵdefineInjectable({ factory: function EnterpriseMetaConfigService_Factory() { return new EnterpriseMetaConfigService(i0.ɵɵinject(i1.MetaConfig)); }, token: EnterpriseMetaConfigService, providedIn: "root" });
    return EnterpriseMetaConfigService;
}());
export { EnterpriseMetaConfigService };
if (false) {
    /** @type {?} */
    EnterpriseMetaConfigService.prototype.attributeGroupList;
    /** @type {?} */
    EnterpriseMetaConfigService.prototype.attributeOptionList;
    /** @type {?} */
    EnterpriseMetaConfigService.prototype.enableForm;
    /** @type {?} */
    EnterpriseMetaConfigService.prototype.attrGroupListSource;
    /** @type {?} */
    EnterpriseMetaConfigService.prototype.currentAttrGroupList;
    /** @type {?} */
    EnterpriseMetaConfigService.prototype.attrOptionListSource;
    /** @type {?} */
    EnterpriseMetaConfigService.prototype.currentAttrOptionList;
    /** @type {?} */
    EnterpriseMetaConfigService.prototype.enableFormSource;
    /** @type {?} */
    EnterpriseMetaConfigService.prototype.currentEnableFormValue;
    /**
     * @type {?}
     * @private
     */
    EnterpriseMetaConfigService.prototype.metaConfigComp;
}
