// @ainosoft/appops-core-components/components/enterprise-meta-config/dist/enterprise-meta-config/esm5/lib/enterprise-meta-config.component.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/enterprise-meta-config.component.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component } from '@angular/core';
import { EnterpriseMetaConfigService } from './enterprise-meta-config.service';
import { MetaConfig } from '../config/MetaConfig';
import { FormBuilder, Validators } from '@angular/forms';
var EnterpriseMetaConfigComponent = /** @class */ (function () {
    function EnterpriseMetaConfigComponent(metaConfig, enterMetaConfigService, _formBuilder) {
        var _this = this;
        this.metaConfig = metaConfig;
        this.enterMetaConfigService = enterMetaConfigService;
        this._formBuilder = _formBuilder;
        this.enableForm = false;
        this.currentSelector = 'Region selector';
        this.stepIndicatorNo = 1;
        this.selectorList = ['Region selector', 'Media selector', 'Service selector', 'Distribution selector', 'SAAS/PAAS selector',
            'Agency discount selector', 'Stream selector', 'Information details', 'Report selector', 'Social selector'];
        this.metaConfig.currentMetaConfigHeader.subscribe((/**
         * @param {?} data
         * @return {?}
         */
        function (data) {
            _this.metaConfigHeader = data;
        }));
        this.enterMetaConfigService.currentAttrGroupList.subscribe((/**
         * @param {?} data
         * @return {?}
         */
        function (data) {
            for (var i = 0; i < data.length; i++) {
                for (var a in data[i]) {
                    _this.attributeGroupList.push(data[i][a]);
                }
            }
            console.log(_this.attributeGroupList);
        }));
        this.enterMetaConfigService.currentAttrOptionList.subscribe((/**
         * @param {?} data
         * @return {?}
         */
        function (data) {
            console.log(data, ' meta option list');
        }));
    }
    /**
     * @return {?}
     */
    EnterpriseMetaConfigComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
        this.firstFormGroup = this._formBuilder.group({
            firstCtrl: ['', Validators.required]
        });
    };
    /**
     * @return {?}
     */
    EnterpriseMetaConfigComponent.prototype.backToSummary = /**
     * @return {?}
     */
    function () {
        this.enterMetaConfigService.enableOpenFormAction(this.enableForm);
    };
    /**
     * @return {?}
     */
    EnterpriseMetaConfigComponent.prototype.prevSelector = /**
     * @return {?}
     */
    function () {
        /** @type {?} */
        var formCard = (/** @type {?} */ (document.getElementsByClassName('form-card-content')[0]));
        formCard.style.animation = 'slide-left 0.5s cubic-bezier(0.25, 0.46, 0.45, 0.94) both';
        for (var i = 0; i < this.attributeGroupList.length; i++) {
            if (this.attributeGroupList[i] === this.currentSelector && i > 0) {
                this.stepIndicatorNo = i;
                i--;
                this.currentSelector = this.attributeGroupList[i];
            }
        }
        setTimeout((/**
         * @return {?}
         */
        function () {
            formCard.style.animation = '';
        }), 500);
    };
    /**
     * @return {?}
     */
    EnterpriseMetaConfigComponent.prototype.nextSelector = /**
     * @return {?}
     */
    function () {
        /** @type {?} */
        var formCard = (/** @type {?} */ (document.getElementsByClassName('form-card-content')[0]));
        formCard.style.animation = 'slide-left 0.5s cubic-bezier(0.25, 0.46, 0.45, 0.94) both';
        for (var i = 0; i < this.attributeGroupList.length; i++) {
            for (var a in this.attributeGroupList[i]) {
                console.log(this.attributeGroupList[i][a]);
                if (a === this.currentSelector && i < this.attributeGroupList.length - 1) {
                    i++;
                    this.currentSelector = this.attributeGroupList[i][a];
                    this.stepIndicatorNo = i + 1;
                    // console.log(this.currentSelector, i);
                }
            }
        }
        setTimeout((/**
         * @return {?}
         */
        function () {
            formCard.style.animation = '';
        }), 500);
    };
    EnterpriseMetaConfigComponent.decorators = [
        { type: Component, args: [{
                    selector: 'lib-enterprise-meta-config',
                    template: "<div class=\"formBgColor\">\n    <mat-toolbar style=\"background: #f0f1f3;\">\n        <button mat-icon-button class=\"back\" (click)=\"backToSummary()\">\n            <mat-icon matTooltip=\"back\">keyboard_backspace</mat-icon>\n        </button>\n        <h2 class=\"meta-config-header\">Configure {{metaConfigHeader}} details</h2>\n    </mat-toolbar>\n\n    <div class=\"form-card-container\">\n        <button mat-fab class=\"step-indicator\">{{stepIndicatorNo}}</button>\n\n        <!-- <svg class=\"progress-circle\" width=\"70\" height=\"70\">\n            <path d=\"m35,2.5c17.955803,0 32.5,14.544199 32.5,32.5c0,17.955803 -14.544197,32.5 -32.5,32.5c-17.955803,0 -32.5,-14.544197 -32.5,-32.5c0,-17.955801 14.544197,-32.5 32.5,-32.5z\"/>\n        </svg> -->\n\n        <div class=\"form-card-content\">\n            <form [formGroup]=\"firstFormGroup\" class=\"form-card\">\n                <mat-card-content>\n                    <p>\n                        <mat-form-field appearance=\"outline\" class=\"name-field\">\n                            <mat-label>{{currentSelector}}</mat-label>\n                            <mat-select formControlName=\"first2Ctrl\" required>\n                                <mat-option>1</mat-option>\n                                <mat-option>2</mat-option>\n                            </mat-select>\n                        </mat-form-field>\n                    </p>\n\n                    <button mat-stroked-button type=\"button\" class=\"prev-button\" (click)=\"prevSelector()\">\n                        <mat-icon>chevron_left</mat-icon>\n                    </button>\n\n                    <button mat-stroked-button type=\"button\" class=\"next-button\" (click)=\"nextSelector()\">\n                        <mat-icon>chevron_right</mat-icon>\n                    </button>\n                </mat-card-content>\n            </form>\n        </div>\n    </div>\n</div>",
                    styles: [".formBgColor{background:#f0f1f3}.meta-config-header{color:#414c55;font-family:Questrial,sans-serif;font-weight:700;letter-spacing:1px}.form-card-container{height:73vh;margin:8px;-webkit-animation:.5s cubic-bezier(.25,.46,.45,.94) both slide-left;animation:.5s cubic-bezier(.25,.46,.45,.94) both slide-left;background:#fff;display:-webkit-box;display:flex}@-webkit-keyframes slide-left{from{-webkit-transform:translateX(150px);transform:translateX(150px)}to{-webkit-transform:translateX(5px);transform:translateX(5px)}}@keyframes slide-left{from{-webkit-transform:translateX(150px);transform:translateX(150px)}to{-webkit-transform:translateX(5px);transform:translateX(5px)}}.form-card-content{background:#f0f1f3;width:50vw;height:50vh;margin-top:5%;box-shadow:0 12px 16px 0 rgba(0,0,0,.24),0 17px 50px 0 rgba(0,0,0,.19)}.form-card{padding:27px}.name-field{width:40%;font-family:Questrial,sans-serif}.next-button{background:#414c55;color:#fff;font-size:16px;outline:0;border:none;padding:5px 40px;font-family:Questrial,sans-serif;float:right;position:relative;bottom:-102px}.next-button:hover{box-shadow:0 12px 16px 0 rgba(0,0,0,.24),0 17px 50px 0 rgba(0,0,0,.19)}.step-indicator{margin:5% 8%;background:#f0f1f3;width:72px;height:72px;font-size:x-large;color:#414c55}.prev-button{background:#414c55;color:#fff;font-size:16px;outline:0;border:none;padding:5px 40px;font-family:Questrial,sans-serif;position:relative;bottom:-102px}.prev-button:hover{box-shadow:0 12px 16px 0 rgba(0,0,0,.24),0 17px 50px 0 rgba(0,0,0,.19)}"]
                }] }
    ];
    /** @nocollapse */
    EnterpriseMetaConfigComponent.ctorParameters = function () { return [
        { type: MetaConfig },
        { type: EnterpriseMetaConfigService },
        { type: FormBuilder }
    ]; };
    return EnterpriseMetaConfigComponent;
}());
export { EnterpriseMetaConfigComponent };
if (false) {
    /** @type {?} */
    EnterpriseMetaConfigComponent.prototype.metaConfigHeader;
    /** @type {?} */
    EnterpriseMetaConfigComponent.prototype.enableForm;
    /** @type {?} */
    EnterpriseMetaConfigComponent.prototype.firstFormGroup;
    /** @type {?} */
    EnterpriseMetaConfigComponent.prototype.currentSelector;
    /** @type {?} */
    EnterpriseMetaConfigComponent.prototype.stepIndicatorNo;
    /** @type {?} */
    EnterpriseMetaConfigComponent.prototype.attributeGroupList;
    /** @type {?} */
    EnterpriseMetaConfigComponent.prototype.selectorList;
    /**
     * @type {?}
     * @private
     */
    EnterpriseMetaConfigComponent.prototype.metaConfig;
    /**
     * @type {?}
     * @private
     */
    EnterpriseMetaConfigComponent.prototype.enterMetaConfigService;
    /**
     * @type {?}
     * @private
     */
    EnterpriseMetaConfigComponent.prototype._formBuilder;
}
