// @ainosoft/appops-core-components/components/enterprise-form/dist/enterprise-form/esm5/lib/enterprise-form.service.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/enterprise-form.service.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Injectable } from '@angular/core';
import { BehaviorSubject } from 'rxjs';
import { FormConfig } from '../config/FormConfig';
import { ActionConfig } from '../config/ActionConfig';
import * as i0 from "@angular/core";
import * as i1 from "../config/FormConfig";
import * as i2 from "../config/ActionConfig";
var EnterpriseFormService = /** @class */ (function () {
    function EnterpriseFormService(formConfigComp, actionConfigComp) {
        this.formConfigComp = formConfigComp;
        this.actionConfigComp = actionConfigComp;
        /**
         * makes the \@currentList as an observable, which emits the current list(updated list)
         *  and can be subscribed by any component.
         */
        this.moduleListSource = new BehaviorSubject(this.list);
        this.currentList = this.moduleListSource.asObservable();
        this.enableGridSource = new BehaviorSubject(this.enableGrid);
        this.currentEnableGridValue = this.enableGridSource.asObservable();
        this.enableMetaConfigSource = new BehaviorSubject(this.enableMetaConfig);
        this.currentEnableMetaConfigValue = this.enableMetaConfigSource.asObservable();
    }
    /** sets the grid list/data provided by the respective external modules(apps/plugins). */
    /**
     * sets the grid list/data provided by the respective external modules(apps/plugins).
     * @param {?} data
     * @return {?}
     */
    EnterpriseFormService.prototype.setFormList = /**
     * sets the grid list/data provided by the respective external modules(apps/plugins).
     * @param {?} data
     * @return {?}
     */
    function (data) {
        this.list = new Array();
        for (var i = 0; i < data.length; i++) {
            this.list.push(data[i]);
        }
        this.moduleListSource.next(this.list);
    };
    /**
     * @param {?} value
     * @return {?}
     */
    EnterpriseFormService.prototype.enableOpenGridAction = /**
     * @param {?} value
     * @return {?}
     */
    function (value) {
        this.enableGrid = value;
        this.enableGridSource.next(this.enableGrid);
    };
    /** sets the form header/name provided from their respective external modules(apps/plugins). */
    /**
     * sets the form header/name provided from their respective external modules(apps/plugins).
     * @param {?} formHeader
     * @return {?}
     */
    EnterpriseFormService.prototype.setFormHeader = /**
     * sets the form header/name provided from their respective external modules(apps/plugins).
     * @param {?} formHeader
     * @return {?}
     */
    function (formHeader) {
        this.formConfigComp.setFormHeader = formHeader;
    };
    /**
     * @param {?} value
     * @return {?}
     */
    EnterpriseFormService.prototype.enableModuleName = /**
     * @param {?} value
     * @return {?}
     */
    function (value) {
        this.actionConfigComp.setModuleNameValue = value;
    };
    /**
     * @param {?} value
     * @return {?}
     */
    EnterpriseFormService.prototype.enableUserRole = /**
     * @param {?} value
     * @return {?}
     */
    function (value) {
        this.actionConfigComp.setUserRoleValue = value;
    };
    /**
     * @param {?} value
     * @return {?}
     */
    EnterpriseFormService.prototype.enableRefPackage = /**
     * @param {?} value
     * @return {?}
     */
    function (value) {
        this.actionConfigComp.setRefPackageValue = value;
        console.log(value);
    };
    /**
     * @param {?} value
     * @return {?}
     */
    EnterpriseFormService.prototype.enableBasePrice = /**
     * @param {?} value
     * @return {?}
     */
    function (value) {
        this.actionConfigComp.setBasePriceValue = value;
    };
    /**
     * @param {?} value
     * @return {?}
     */
    EnterpriseFormService.prototype.enableSubFrequency = /**
     * @param {?} value
     * @return {?}
     */
    function (value) {
        this.actionConfigComp.setSubFrequencyValue = value;
    };
    /**
     * @param {?} value
     * @return {?}
     */
    EnterpriseFormService.prototype.enableOpenMetaConfigAction = /**
     * @param {?} value
     * @return {?}
     */
    function (value) {
        this.enableMetaConfig = value;
        this.enableMetaConfigSource.next(this.enableMetaConfig);
    };
    EnterpriseFormService.decorators = [
        { type: Injectable, args: [{
                    providedIn: 'root'
                },] }
    ];
    /** @nocollapse */
    EnterpriseFormService.ctorParameters = function () { return [
        { type: FormConfig },
        { type: ActionConfig }
    ]; };
    /** @nocollapse */ EnterpriseFormService.ngInjectableDef = i0.ɵɵdefineInjectable({ factory: function EnterpriseFormService_Factory() { return new EnterpriseFormService(i0.ɵɵinject(i1.FormConfig), i0.ɵɵinject(i2.ActionConfig)); }, token: EnterpriseFormService, providedIn: "root" });
    return EnterpriseFormService;
}());
export { EnterpriseFormService };
if (false) {
    /** @type {?} */
    EnterpriseFormService.prototype.list;
    /** @type {?} */
    EnterpriseFormService.prototype.enableGrid;
    /** @type {?} */
    EnterpriseFormService.prototype.enableMetaConfig;
    /**
     * makes the \@currentList as an observable, which emits the current list(updated list)
     *  and can be subscribed by any component.
     * @type {?}
     */
    EnterpriseFormService.prototype.moduleListSource;
    /** @type {?} */
    EnterpriseFormService.prototype.currentList;
    /** @type {?} */
    EnterpriseFormService.prototype.enableGridSource;
    /** @type {?} */
    EnterpriseFormService.prototype.currentEnableGridValue;
    /** @type {?} */
    EnterpriseFormService.prototype.enableMetaConfigSource;
    /** @type {?} */
    EnterpriseFormService.prototype.currentEnableMetaConfigValue;
    /**
     * @type {?}
     * @private
     */
    EnterpriseFormService.prototype.formConfigComp;
    /**
     * @type {?}
     * @private
     */
    EnterpriseFormService.prototype.actionConfigComp;
}
