// @ainosoft/appops-core-components/components/enterprise-form/dist/enterprise-form/esm5/lib/enterprise-form.component.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/enterprise-form.component.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component } from '@angular/core';
import { FormBuilder, Validators } from '@angular/forms';
import { EnterpriseFormService } from './enterprise-form.service';
import { FormConfig } from '../config/FormConfig';
import { ActionConfig } from '../config/ActionConfig';
var EnterpriseFormComponent = /** @class */ (function () {
    function EnterpriseFormComponent(_formBuilder, enterpFormService, formConfig, actionConfig) {
        var _this = this;
        this._formBuilder = _formBuilder;
        this.enterpFormService = enterpFormService;
        this.formConfig = formConfig;
        this.actionConfig = actionConfig;
        this.isEditable = false;
        this.isLinear = true;
        this.enableGrid = false;
        this.referencePackageList = [];
        this.enableModuleName = false;
        this.enableUserRole = false;
        this.enableRefPackage = false;
        this.enableBasePrice = false;
        this.enableSubFrequency = false;
        this.enableMetaConfig = false;
        /** subscribes to the last emitted grid list (current list) from the behaviour subject
         *  @currentList observable instance.
         */
        this.enterpFormService.currentList.subscribe((/**
         * @param {?} data
         * @return {?}
         */
        function (data) {
            console.log(data, 'form moduleList');
            for (var i in data) {
                if (data[i].status === 'ready' || data[i].status === 'subscribed') {
                    _this.referencePackageList.push(data[i].packageName);
                }
            }
        }));
        this.formConfig.currentFormHeader.subscribe((/**
         * @param {?} data
         * @return {?}
         */
        function (data) {
            _this.formHeader = data;
        }));
        this.actionConfig.currModuleNameValue.subscribe((/**
         * @param {?} data
         * @return {?}
         */
        function (data) {
            _this.enableModuleName = data;
        }));
        this.actionConfig.currUserRoleValue.subscribe((/**
         * @param {?} data
         * @return {?}
         */
        function (data) {
            _this.enableUserRole = data;
        }));
        this.actionConfig.currRefPackageValue.subscribe((/**
         * @param {?} data
         * @return {?}
         */
        function (data) {
            _this.enableRefPackage = data;
        }));
        this.actionConfig.curreBasePriceValue.subscribe((/**
         * @param {?} data
         * @return {?}
         */
        function (data) {
            _this.enableBasePrice = data;
        }));
        this.actionConfig.curreSubFrequencyValue.subscribe((/**
         * @param {?} data
         * @return {?}
         */
        function (data) {
            _this.enableSubFrequency = data;
        }));
    }
    /**
     * @return {?}
     */
    EnterpriseFormComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
        this.firstFormGroup = this._formBuilder.group({
            firstCtrl: ['', Validators.required],
            first2Ctrl: ['', Validators.required],
        });
        this.secondFormGroup = this._formBuilder.group({
            secondCtrl: ['', Validators.required]
        });
    };
    /**
     * @return {?}
     */
    EnterpriseFormComponent.prototype.backToTable = /**
     * @return {?}
     */
    function () {
        this.enableGrid = true;
        this.enterpFormService.enableOpenGridAction(this.enableGrid);
    };
    /**
     * @return {?}
     */
    EnterpriseFormComponent.prototype.openMetaConfig = /**
     * @return {?}
     */
    function () {
        this.enableMetaConfig = true;
        this.enterpFormService.enableOpenMetaConfigAction(this.enableMetaConfig);
    };
    EnterpriseFormComponent.decorators = [
        { type: Component, args: [{
                    selector: 'lib-enterprise-form',
                    template: "<div class=\"formBgColor\">\n    <mat-toolbar style=\"background: #f0f1f3;\">\n        <button mat-icon-button class=\"back\" (click)=\"backToTable()\">\n            <mat-icon matTooltip=\"back\">keyboard_backspace</mat-icon>\n        </button>\n        <h2 class=\"form-header\">Create new {{formHeader | lowercase }}</h2>\n    </mat-toolbar>\n\n    <div class=\"form-card-container\">\n        <form [formGroup]=\"firstFormGroup\" class=\"form-card\">\n            <h2 class=\"meta-header\">Summary</h2>\n            <mat-card-content>\n                <!-- Module Name -->\n                <p *ngIf=\"enableModuleName\">\n                    <mat-form-field appearance=\"outline\" class=\"name-field\">\n                        <mat-label>Name</mat-label>\n                        <input matInput placeholder=\"Name\" formControlName=\"firstCtrl\" required>\n                    </mat-form-field>\n                </p>\n\n                <!-- User Role -->\n                <p *ngIf=\"enableUserRole\">\n                    <mat-form-field appearance=\"outline\" class=\"select-field\">\n                        <mat-label>{{formHeader }} role</mat-label>\n                        <mat-select formControlName=\"first2Ctrl\">\n                            <mat-option value=\"Queries\">role 1</mat-option>\n                            <mat-option value=\"Items\">role 2</mat-option>\n                        </mat-select>\n                    </mat-form-field>\n                </p>\n\n                <!-- User refrence package -->\n                <p *ngIf=\"enableRefPackage\">\n                    <mat-form-field appearance=\"outline\" class=\"select-field\">\n                        <mat-label>Use reference {{formHeader | lowercase }}</mat-label>\n                        <mat-select formControlName=\"first2Ctrl\" required>\n                            <mat-option>--select--</mat-option>\n                            <mat-option value=\"{{element}}\" *ngFor='let element of referencePackageList' >{{element}}</mat-option>\n                        </mat-select>\n                    </mat-form-field>\n                </p>\n\n                <!-- Base Price -->\n                <p *ngIf=\"enableBasePrice\">\n                    <mat-form-field appearance=\"outline\" class=\"name-field\">\n                        <mat-label>Base price</mat-label>\n                        <input matInput placeholder=\"Base price\" formControlName=\"firstCtrl\" required>\n                        <i class=\"fa fa-euro\" style=\"font-size:22px\" matSuffix></i>\n                    </mat-form-field>\n                </p>\n\n                <!-- Subscription frequency -->\n                <p *ngIf=\"enableSubFrequency\">\n                    <mat-form-field appearance=\"outline\" class=\"select-field\">\n                        <mat-label>Subscription frequency</mat-label>\n                        <mat-select formControlName=\"first2Ctrl\">\n                            <mat-option value=\"Quaterly\">Monthly</mat-option>\n                            <mat-option value=\"Quaterly\">Quaterly</mat-option>\n                            <mat-option value=\"Half Year\">Half Yearly</mat-option>\n                            <mat-option value=\"Annually\">Annually</mat-option>\n                        </mat-select>\n                    </mat-form-field>\n                </p>\n                <button mat-stroked-button type=\"button\" class=\"next-button\" (click)=\"openMetaConfig()\">{{formHeader | lowercase}} details\n                    <mat-icon>chevron_right</mat-icon>\n                </button>\n\n            </mat-card-content>\n        </form>\n    </div>\n</div>",
                    styles: [".formBgColor{background:#f0f1f3}.form-card-container{margin:8px;height:73vh;-webkit-animation:.5s cubic-bezier(.25,.46,.45,.94) both slide-left;animation:.5s cubic-bezier(.25,.46,.45,.94) both slide-left;background:#fff}@-webkit-keyframes slide-left{from{transform:translateX(150px)}to{transform:translateX(5px)}}@keyframes slide-left{from{transform:translateX(150px)}to{transform:translateX(5px)}}.form-header{color:#414c55;font-family:Questrial,sans-serif;font-weight:700;letter-spacing:1px}.meta-header{color:#414c55;font-family:Questrial,sans-serif;letter-spacing:1px;font-weight:100}.form-card{padding:27px}.mat-option{font-size:smaller}.name-field,.select-field{width:40%;font-family:Questrial,sans-serif}.next-button{position:absolute;bottom:30px;right:50px;background:#414c55;color:#fff;font-size:16px;outline:0;border:none;padding:5px 40px;font-family:Questrial,sans-serif}.next-button:hover{box-shadow:0 12px 16px 0 rgba(0,0,0,.24),0 17px 50px 0 rgba(0,0,0,.19)}"]
                }] }
    ];
    /** @nocollapse */
    EnterpriseFormComponent.ctorParameters = function () { return [
        { type: FormBuilder },
        { type: EnterpriseFormService },
        { type: FormConfig },
        { type: ActionConfig }
    ]; };
    return EnterpriseFormComponent;
}());
export { EnterpriseFormComponent };
if (false) {
    /** @type {?} */
    EnterpriseFormComponent.prototype.firstFormGroup;
    /** @type {?} */
    EnterpriseFormComponent.prototype.secondFormGroup;
    /** @type {?} */
    EnterpriseFormComponent.prototype.isEditable;
    /** @type {?} */
    EnterpriseFormComponent.prototype.isLinear;
    /** @type {?} */
    EnterpriseFormComponent.prototype.enableGrid;
    /** @type {?} */
    EnterpriseFormComponent.prototype.formHeader;
    /** @type {?} */
    EnterpriseFormComponent.prototype.referencePackageList;
    /** @type {?} */
    EnterpriseFormComponent.prototype.enableModuleName;
    /** @type {?} */
    EnterpriseFormComponent.prototype.enableUserRole;
    /** @type {?} */
    EnterpriseFormComponent.prototype.enableRefPackage;
    /** @type {?} */
    EnterpriseFormComponent.prototype.enableBasePrice;
    /** @type {?} */
    EnterpriseFormComponent.prototype.enableSubFrequency;
    /** @type {?} */
    EnterpriseFormComponent.prototype.enableMetaConfig;
    /**
     * @type {?}
     * @private
     */
    EnterpriseFormComponent.prototype._formBuilder;
    /**
     * @type {?}
     * @private
     */
    EnterpriseFormComponent.prototype.enterpFormService;
    /**
     * @type {?}
     * @private
     */
    EnterpriseFormComponent.prototype.formConfig;
    /**
     * @type {?}
     * @private
     */
    EnterpriseFormComponent.prototype.actionConfig;
}
