// @ainosoft/appops-core-components/components/enterprise-form/dist/enterprise-form/esm2015/lib/enterprise-form.module.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/enterprise-form.module.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { NgModule, CUSTOM_ELEMENTS_SCHEMA, NO_ERRORS_SCHEMA } from '@angular/core';
import { EnterpriseFormComponent } from './enterprise-form.component';
import { CommonModule } from '@angular/common';
import { EnterpriseFormService } from './enterprise-form.service';
import { MaterialModuleSet } from '../common/material-module';
export class EnterpriseFormModule {
}
EnterpriseFormModule.decorators = [
    { type: NgModule, args: [{
                declarations: [EnterpriseFormComponent],
                imports: [
                    MaterialModuleSet,
                    CommonModule,
                ],
                exports: [EnterpriseFormComponent],
                providers: [EnterpriseFormService],
                schemas: [
                    CUSTOM_ELEMENTS_SCHEMA, NO_ERRORS_SCHEMA
                ],
            },] }
];
