// @ainosoft/appops-core-components/components/enterprise-form/dist/enterprise-form/esm5/config/ActionConfig.js
/**
 * @fileoverview added by tsickle
 * Generated from: config/ActionConfig.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Injectable } from "@angular/core";
import { BehaviorSubject } from "rxjs";
import * as i0 from "@angular/core";
var ActionConfig = /** @class */ (function () {
    function ActionConfig() {
        /**
         * makes the \@currCreateIconValue as an observable, it emits the current boolean
         *  value for create icon, which can be subscribed by any component.
         */
        this.moduleNameSource = new BehaviorSubject(this.enableModuleName);
        this.currModuleNameValue = this.moduleNameSource.asObservable();
        this.userRoleSource = new BehaviorSubject(this.enableUserRole);
        this.currUserRoleValue = this.userRoleSource.asObservable();
        this.refPackageSource = new BehaviorSubject(this.enableRefPackage);
        this.currRefPackageValue = this.refPackageSource.asObservable();
        this.basePriceSource = new BehaviorSubject(this.enableBasePrice);
        this.curreBasePriceValue = this.basePriceSource.asObservable();
        this.subFrequencySource = new BehaviorSubject(this.enableSubFrequency);
        this.curreSubFrequencyValue = this.subFrequencySource.asObservable();
    }
    Object.defineProperty(ActionConfig.prototype, "getModuleNameValue", {
        get: /**
         * @return {?}
         */
        function () {
            return this.enableModuleName;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(ActionConfig.prototype, "setModuleNameValue", {
        set: /**
         * @param {?} value
         * @return {?}
         */
        function (value) {
            this.enableModuleName = value;
            this.moduleNameSource.next(this.enableModuleName);
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(ActionConfig.prototype, "getUserRoleValue", {
        get: /**
         * @return {?}
         */
        function () {
            return this.enableUserRole;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(ActionConfig.prototype, "setUserRoleValue", {
        set: /**
         * @param {?} value
         * @return {?}
         */
        function (value) {
            this.enableUserRole = value;
            this.userRoleSource.next(this.enableUserRole);
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(ActionConfig.prototype, "getRefPackageValue", {
        get: /**
         * @return {?}
         */
        function () {
            return this.enableRefPackage;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(ActionConfig.prototype, "setRefPackageValue", {
        set: /**
         * @param {?} value
         * @return {?}
         */
        function (value) {
            this.enableRefPackage = value;
            this.refPackageSource.next(this.enableRefPackage);
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(ActionConfig.prototype, "getBasePriceValue", {
        get: /**
         * @return {?}
         */
        function () {
            return this.enableBasePrice;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(ActionConfig.prototype, "setBasePriceValue", {
        set: /**
         * @param {?} value
         * @return {?}
         */
        function (value) {
            this.enableBasePrice = value;
            this.basePriceSource.next(this.enableBasePrice);
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(ActionConfig.prototype, "getSubFrequencyValue", {
        get: /**
         * @return {?}
         */
        function () {
            return this.enableSubFrequency;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(ActionConfig.prototype, "setSubFrequencyValue", {
        set: /**
         * @param {?} value
         * @return {?}
         */
        function (value) {
            this.enableSubFrequency = value;
            this.subFrequencySource.next(this.enableSubFrequency);
        },
        enumerable: true,
        configurable: true
    });
    ActionConfig.decorators = [
        { type: Injectable, args: [{
                    providedIn: 'root'
                },] }
    ];
    /** @nocollapse */ ActionConfig.ngInjectableDef = i0.ɵɵdefineInjectable({ factory: function ActionConfig_Factory() { return new ActionConfig(); }, token: ActionConfig, providedIn: "root" });
    return ActionConfig;
}());
export { ActionConfig };
if (false) {
    /** @type {?} */
    ActionConfig.prototype.enableModuleName;
    /** @type {?} */
    ActionConfig.prototype.enableUserRole;
    /** @type {?} */
    ActionConfig.prototype.enableRefPackage;
    /** @type {?} */
    ActionConfig.prototype.enableBasePrice;
    /** @type {?} */
    ActionConfig.prototype.enableSubFrequency;
    /**
     * makes the \@currCreateIconValue as an observable, it emits the current boolean
     *  value for create icon, which can be subscribed by any component.
     * @type {?}
     */
    ActionConfig.prototype.moduleNameSource;
    /** @type {?} */
    ActionConfig.prototype.currModuleNameValue;
    /** @type {?} */
    ActionConfig.prototype.userRoleSource;
    /** @type {?} */
    ActionConfig.prototype.currUserRoleValue;
    /** @type {?} */
    ActionConfig.prototype.refPackageSource;
    /** @type {?} */
    ActionConfig.prototype.currRefPackageValue;
    /** @type {?} */
    ActionConfig.prototype.basePriceSource;
    /** @type {?} */
    ActionConfig.prototype.curreBasePriceValue;
    /** @type {?} */
    ActionConfig.prototype.subFrequencySource;
    /** @type {?} */
    ActionConfig.prototype.curreSubFrequencyValue;
}
