// @ainosoft/appops-core-components/components/enterprise-form/dist/enterprise-form/esm5/config/FormConfig.js
/**
 * @fileoverview added by tsickle
 * Generated from: config/FormConfig.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Injectable } from "@angular/core";
import { BehaviorSubject } from "rxjs/internal/BehaviorSubject";
import * as i0 from "@angular/core";
var FormConfig = /** @class */ (function () {
    function FormConfig() {
        /**
         * makes the \@currentFormHeader as an observable, which emits the current form header(updated header)
         *   and can be subscribed by any component.
         */
        this.formHeaderSource = new BehaviorSubject(this.formHeader);
        this.currentFormHeader = this.formHeaderSource.asObservable();
    }
    Object.defineProperty(FormConfig.prototype, "getFormHeader", {
        get: /**
         * @return {?}
         */
        function () {
            return this.formHeader;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(FormConfig.prototype, "setFormHeader", {
        set: /**
         * @param {?} formHeader
         * @return {?}
         */
        function (formHeader) {
            this.formHeader = formHeader;
            this.formHeaderSource.next(this.formHeader);
        },
        enumerable: true,
        configurable: true
    });
    FormConfig.decorators = [
        { type: Injectable, args: [{
                    providedIn: 'root'
                },] }
    ];
    /** @nocollapse */ FormConfig.ngInjectableDef = i0.ɵɵdefineInjectable({ factory: function FormConfig_Factory() { return new FormConfig(); }, token: FormConfig, providedIn: "root" });
    return FormConfig;
}());
export { FormConfig };
if (false) {
    /** @type {?} */
    FormConfig.prototype.formHeader;
    /**
     * makes the \@currentFormHeader as an observable, which emits the current form header(updated header)
     *   and can be subscribed by any component.
     * @type {?}
     */
    FormConfig.prototype.formHeaderSource;
    /** @type {?} */
    FormConfig.prototype.currentFormHeader;
}
