// @ainosoft/appops-core-components/components/enterprise-form/src/config/ActionConfig.ts
import { Injectable } from "@angular/core";
import { BehaviorSubject } from "rxjs";

@Injectable({
    providedIn: 'root'
})
export class ActionConfig {
    
    enableModuleName: boolean;
    enableUserRole: boolean;
    enableRefPackage: boolean;
    enableBasePrice: boolean;
    enableSubFrequency: boolean;

    /** makes the @currCreateIconValue as an observable, it emits the current boolean 
     *  value for create icon, which can be subscribed by any component.
    */
    public moduleNameSource = new BehaviorSubject<boolean>(this.enableModuleName);
    currModuleNameValue = this.moduleNameSource.asObservable();

    public userRoleSource = new BehaviorSubject<boolean>(this.enableUserRole);
    currUserRoleValue = this.userRoleSource.asObservable();

    public refPackageSource = new BehaviorSubject<boolean>(this.enableRefPackage);
    currRefPackageValue = this.refPackageSource.asObservable();

    public basePriceSource = new BehaviorSubject<boolean>(this.enableBasePrice);
    curreBasePriceValue = this.basePriceSource.asObservable();

    public subFrequencySource = new BehaviorSubject<boolean>(this.enableSubFrequency);
    curreSubFrequencyValue = this.subFrequencySource.asObservable();

    get getModuleNameValue() {
        return this.enableModuleName;
    }

    set setModuleNameValue(value: boolean) {
        this.enableModuleName = value;
        this.moduleNameSource.next(this.enableModuleName);
    }

    get getUserRoleValue() {
        return this.enableUserRole;
    }

    set setUserRoleValue(value: boolean) {
        this.enableUserRole = value;
        this.userRoleSource.next(this.enableUserRole);
    }

    get getRefPackageValue() {
        return this.enableRefPackage;
    }

    set setRefPackageValue(value: boolean) {
        this.enableRefPackage = value;
        this.refPackageSource.next(this.enableRefPackage);
    }

    get getBasePriceValue() {
        return this.enableBasePrice;
    }

    set setBasePriceValue(value: boolean) {
        this.enableBasePrice = value;
        this.basePriceSource.next(this.enableBasePrice);
    }

    get getSubFrequencyValue() {
        return this.enableSubFrequency;
    }

    set setSubFrequencyValue(value: boolean) {
        this.enableSubFrequency = value;
        this.subFrequencySource.next(this.enableSubFrequency);
    }
}
