// @ainosoft/appops-core-components/components/enterprise-meta-config/src/lib/enterprise-meta-config.component.ts
import { Component, OnInit } from '@angular/core';
import { EnterpriseMetaConfigService } from './enterprise-meta-config.service';
import { MetaConfig } from '../config/MetaConfig';
import { FormGroup, FormBuilder, Validators } from '@angular/forms';

@Component({
  selector: 'lib-enterprise-meta-config',
  templateUrl: './enterprise-meta-config.component.html',
  styleUrls: ['./enterprise-meta-config.component.scss'],
})
export class EnterpriseMetaConfigComponent implements OnInit {

  metaConfigHeader: string;
  enableForm = false;
  firstFormGroup: FormGroup;

  currentSelector: string = 'Region Selector';
  currentOptions = [];

  stepIndicatorNo: number = 1;

  attributeGrpAndOptsList;
  selectorList = [];
  optionList = [];

  constructor(private metaConfig: MetaConfig, private enterMetaConfigService: EnterpriseMetaConfigService, private _formBuilder: FormBuilder) { 

    this.metaConfig.currentMetaConfigHeader.subscribe(data => {
      this.metaConfigHeader = data;
    });

    this.enterMetaConfigService.currentAttrGrpAndOptsList.subscribe(data => {
      console.log(data, ' g and opts list')
      this.attributeGrpAndOptsList = data;
      
    });
  }

  ngOnInit() {
    this.firstFormGroup = this._formBuilder.group({
      firstCtrl: ['', Validators.required]
    });
  }

  backToSummary() {
    this.enterMetaConfigService.enableOpenFormAction(this.enableForm);
  }

  prevSelector() {
    let formCard = document.getElementsByClassName('form-card-content')[0] as HTMLElement;
    formCard.style.animation = 'slide-left 0.5s cubic-bezier(0.25, 0.46, 0.45, 0.94) both';

    // for(let i=0; i< this.selectorList.length; i++) {
    //   if(this.selectorList[i] === this.currentSelector && i > 0) {
    //     this.stepIndicatorNo = i;
    //     i--;
    //     this.currentSelector = this.selectorList[i];
    //   }
    // }

    console.log(this.currentSelector);
    setTimeout(function() {
      formCard.style.animation = '';
    }, 500);
  }

  nextSelector() {
    let formCard = document.getElementsByClassName('form-card-content')[0] as HTMLElement;
    formCard.style.animation = 'slide-left 0.5s cubic-bezier(0.25, 0.46, 0.45, 0.94) both';
    
    // for(let i=0; i< this.selectorList.length; i++) {

    //   if(this.selectorList[i] === this.currentSelector && i < (this.attributeGroupList.length - 1)) {
    //     i++;
    //     this.currentSelector = this.selectorList[i];
    //     this.stepIndicatorNo = i+1;
    //     console.log(this.currentSelector, ' ', this.selectorList.length, ' ', i);
    //   }
    // }

    setTimeout(function() {
      formCard.style.animation = '';
    }, 500);
  }

}
