// @ainosoft/appops-core-components/components/enterprise-meta-config/src/lib/enterprise-meta-config.service.ts
import { Injectable } from '@angular/core';
import { MetaConfig } from '../config/MetaConfig';
import { BehaviorSubject } from 'rxjs';

@Injectable({
  providedIn: 'root'
})
export class EnterpriseMetaConfigService {

  attributeGroupList: String[] = [];
  attributeOptionList: String[];
  attrGrpAndOptsList: String[] = [];
  enableForm: boolean;

  constructor(private metaConfigComp?: MetaConfig) { }

  public attrGroupListSource = new BehaviorSubject<any[]>(this.attributeGroupList);
  currentAttrGroupList = this.attrGroupListSource.asObservable();

  public attrOptionListSource = new BehaviorSubject<any[]>(this.attributeOptionList);
  currentAttrOptionList = this.attrOptionListSource.asObservable();

  public enableFormSource = new BehaviorSubject<boolean>(this.enableForm);
  currentEnableFormValue = this.enableFormSource.asObservable();
  
  public attrGrpAndOptsListSource = new BehaviorSubject<any[]>(this.attrGrpAndOptsList);
  currentAttrGrpAndOptsList = this.attrGrpAndOptsListSource.asObservable();

  enableOpenFormAction(value: boolean) {
    this.enableForm = value;
    this.enableFormSource.next(this.enableForm);
  }
  
  setMetaConfigHeader(metaConfigHeader: string) {
    this.metaConfigComp.setMetaConfigHeader = metaConfigHeader;
  }

  setAttrGroupList(data: any[]) {
    this.attributeGroupList = new Array();
    for(let i=0; i<data.length; i++) {
      this.attributeGroupList.push(data[i]);
    }
    this.attrGroupListSource.next(this.attributeGroupList);
  }

  setAttrOptionList(data: any[]) {
    this.attributeOptionList = new Array();
    for(let i=0; i<data.length; i++) {
      this.attributeOptionList.push(data[i]);
    }
    this.attrOptionListSource.next(this.attributeOptionList);
  }

  setAttrGrpAndOptsList(data: any[]) {
    this.attrGrpAndOptsList = new Array();
    for(let i=0; i<data.length; i++) {
      this.attrGrpAndOptsList.push(data[i]);
    }
    this.attrGrpAndOptsListSource.next(this.attrGrpAndOptsList);
  }
}
