// @ainosoft/appops-core-components/components/enterprise-form/dist/enterprise-form/esm2015/config/ActionConfig.js
/**
 * @fileoverview added by tsickle
 * Generated from: config/ActionConfig.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Injectable } from "@angular/core";
import { BehaviorSubject } from "rxjs";
import * as i0 from "@angular/core";
export class ActionConfig {
    constructor() {
        /**
         * makes the \@currCreateIconValue as an observable, it emits the current boolean
         *  value for create icon, which can be subscribed by any component.
         */
        this.moduleNameSource = new BehaviorSubject(this.enableModuleName);
        this.currModuleNameValue = this.moduleNameSource.asObservable();
        this.userRoleSource = new BehaviorSubject(this.enableUserRole);
        this.currUserRoleValue = this.userRoleSource.asObservable();
        this.refPackageSource = new BehaviorSubject(this.enableRefPackage);
        this.currRefPackageValue = this.refPackageSource.asObservable();
        this.basePriceSource = new BehaviorSubject(this.enableBasePrice);
        this.curreBasePriceValue = this.basePriceSource.asObservable();
        this.subFrequencySource = new BehaviorSubject(this.enableSubFrequency);
        this.curreSubFrequencyValue = this.subFrequencySource.asObservable();
    }
    /**
     * @return {?}
     */
    get getModuleNameValue() {
        return this.enableModuleName;
    }
    /**
     * @param {?} value
     * @return {?}
     */
    set setModuleNameValue(value) {
        this.enableModuleName = value;
        this.moduleNameSource.next(this.enableModuleName);
    }
    /**
     * @return {?}
     */
    get getUserRoleValue() {
        return this.enableUserRole;
    }
    /**
     * @param {?} value
     * @return {?}
     */
    set setUserRoleValue(value) {
        this.enableUserRole = value;
        this.userRoleSource.next(this.enableUserRole);
    }
    /**
     * @return {?}
     */
    get getRefPackageValue() {
        return this.enableRefPackage;
    }
    /**
     * @param {?} value
     * @return {?}
     */
    set setRefPackageValue(value) {
        this.enableRefPackage = value;
        this.refPackageSource.next(this.enableRefPackage);
    }
    /**
     * @return {?}
     */
    get getBasePriceValue() {
        return this.enableBasePrice;
    }
    /**
     * @param {?} value
     * @return {?}
     */
    set setBasePriceValue(value) {
        this.enableBasePrice = value;
        this.basePriceSource.next(this.enableBasePrice);
    }
    /**
     * @return {?}
     */
    get getSubFrequencyValue() {
        return this.enableSubFrequency;
    }
    /**
     * @param {?} value
     * @return {?}
     */
    set setSubFrequencyValue(value) {
        this.enableSubFrequency = value;
        this.subFrequencySource.next(this.enableSubFrequency);
    }
}
ActionConfig.decorators = [
    { type: Injectable, args: [{
                providedIn: 'root'
            },] }
];
/** @nocollapse */ ActionConfig.ngInjectableDef = i0.ɵɵdefineInjectable({ factory: function ActionConfig_Factory() { return new ActionConfig(); }, token: ActionConfig, providedIn: "root" });
if (false) {
    /** @type {?} */
    ActionConfig.prototype.enableModuleName;
    /** @type {?} */
    ActionConfig.prototype.enableUserRole;
    /** @type {?} */
    ActionConfig.prototype.enableRefPackage;
    /** @type {?} */
    ActionConfig.prototype.enableBasePrice;
    /** @type {?} */
    ActionConfig.prototype.enableSubFrequency;
    /**
     * makes the \@currCreateIconValue as an observable, it emits the current boolean
     *  value for create icon, which can be subscribed by any component.
     * @type {?}
     */
    ActionConfig.prototype.moduleNameSource;
    /** @type {?} */
    ActionConfig.prototype.currModuleNameValue;
    /** @type {?} */
    ActionConfig.prototype.userRoleSource;
    /** @type {?} */
    ActionConfig.prototype.currUserRoleValue;
    /** @type {?} */
    ActionConfig.prototype.refPackageSource;
    /** @type {?} */
    ActionConfig.prototype.currRefPackageValue;
    /** @type {?} */
    ActionConfig.prototype.basePriceSource;
    /** @type {?} */
    ActionConfig.prototype.curreBasePriceValue;
    /** @type {?} */
    ActionConfig.prototype.subFrequencySource;
    /** @type {?} */
    ActionConfig.prototype.curreSubFrequencyValue;
}
