// @ainosoft/appops-core-components/components/enterprise-meta-config/dist/enterprise-meta-config/esm2015/config/MetaConfig.js
/**
 * @fileoverview added by tsickle
 * Generated from: config/MetaConfig.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Injectable } from "@angular/core";
import { BehaviorSubject } from "rxjs/internal/BehaviorSubject";
import * as i0 from "@angular/core";
export class MetaConfig {
    constructor() {
        /**
         * makes the \@currentFormHeader as an observable, which emits the current form header(updated header)
         *   and can be subscribed by any component.
         */
        this.metaConfigHeaderSource = new BehaviorSubject(this.metaConfigHeader);
        this.currentMetaConfigHeader = this.metaConfigHeaderSource.asObservable();
    }
    /**
     * @return {?}
     */
    get getMetaConfigHeader() {
        return this.metaConfigHeader;
    }
    /**
     * @param {?} metaConfigHeader
     * @return {?}
     */
    set setMetaConfigHeader(metaConfigHeader) {
        this.metaConfigHeader = metaConfigHeader;
        this.metaConfigHeaderSource.next(this.metaConfigHeader);
    }
}
MetaConfig.decorators = [
    { type: Injectable, args: [{
                providedIn: 'root'
            },] }
];
/** @nocollapse */ MetaConfig.ngInjectableDef = i0.ɵɵdefineInjectable({ factory: function MetaConfig_Factory() { return new MetaConfig(); }, token: MetaConfig, providedIn: "root" });
if (false) {
    /** @type {?} */
    MetaConfig.prototype.metaConfigHeader;
    /**
     * makes the \@currentFormHeader as an observable, which emits the current form header(updated header)
     *   and can be subscribed by any component.
     * @type {?}
     */
    MetaConfig.prototype.metaConfigHeaderSource;
    /** @type {?} */
    MetaConfig.prototype.currentMetaConfigHeader;
}
