// @ainosoft/appops-core-components/components/enterprise-grid/dist/enterprise-grid/esm5/lib/enterprise-grid.module.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/enterprise-grid.module.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { NgModule, CUSTOM_ELEMENTS_SCHEMA, NO_ERRORS_SCHEMA } from '@angular/core';
import { EnterpriseGridComponent } from './enterprise-grid.component';
import { CommonModule } from '@angular/common';
import { MaterialModuleSet } from '../common/material-module';
import { AoGridDataDirective } from '../directive/ao-grid-data.directive';
import { AoActionsDirective } from '../directive/ao-actions.directive';
import { AoGridConfigDirective } from '../directive/ao-grid-config.directive';
var EnterpriseGridModule = /** @class */ (function () {
    function EnterpriseGridModule() {
    }
    EnterpriseGridModule.decorators = [
        { type: NgModule, args: [{
                    declarations: [EnterpriseGridComponent, AoGridDataDirective, AoActionsDirective, AoGridConfigDirective],
                    imports: [
                        CommonModule,
                        MaterialModuleSet,
                    ],
                    exports: [EnterpriseGridComponent, AoGridDataDirective, AoActionsDirective, AoGridConfigDirective],
                    providers: [EnterpriseGridComponent],
                    schemas: [
                        CUSTOM_ELEMENTS_SCHEMA, NO_ERRORS_SCHEMA
                    ],
                },] }
    ];
    return EnterpriseGridModule;
}());
export { EnterpriseGridModule };
